src/helpers/library.ts

```diff
@@ -300,7 +300,7 @@ function getRecentGames(libraries: GameInfo[]) {
     (configStore.get('games.recent', []) as Array<RecentGame>) || []
   return recentGames.filter(Boolean).map((game) => {
     return libraries.find((info) => info.app_name === game.appName)
-  }) as GameInfo[] // With this `.filter`, no empty entries can be returned. TS doesn't understand this, so we have to add the `as GameInfo[]`
+  }) as GameInfo[] // needed to return empty array instead undefined
```

Comment:
Hm, now that I see it, is this `as GameInfo[]` even right anymore? Don't we need *another* `.filter()` to make sure no `undefined` can get into the list?

Reply:
See filter above (`.filter(Boolean)`). This filter was already there before but wrong placed. It should be before `libraries.find`, not after.

Reply:
I'm aware the `.filter` was there already, I wrote the code. What I'm saying is, if `libraries.find(...)` returns `undefined` (as far as I know that's what it does if no element is found), wouldn't we then return an array of `GameInfo | undefined`, not just `GameInfo`?

As far as I understand, this code maps recent games (which are stored locally) to their respective GameInfo. What if you had a game that is for some reason no longer in your account (for example, you refunded it)? Wouldn't that then result in the config store still having the recent game stored, but the `libraries` not including it anymore? So wouldn't that then mean the second `.filter(...)` *is* necessary?

Reply:
Maybe, but this is not the topic of this PR. The PR is about to fix the empty screen and not to take care of refunded games. Also from my understanding `libraries.find` does actually compare recent against the libraries collection. A removed game should not be in the library on restart of heroic. So the fix should be on another place, if it is a problem.

The GameInfo[] | undefined was a problem also before. Will see how i can solve that.

Reply:
Ok i checked back. The approach before was correct.
A undefined[] array is still a object but with length 0. Casting it to an empty GameInfo[] array is allowed.

Reply:
> Maybe, but this is not the topic of this PR

That was the point of the `.filter(Boolean)` before. By moving it elsewhere, you're removing that fix

> The GameInfo[] | undefined was a problem also before. Will see how i can solve that.

No, with the `.filter(Boolean)` it wasn't, since `undefined` is filtered out (this is also not quite the right terminology, so just so we understand each other, it's `(GameInfo | undefined)[]` (so an array whose members are either GameInfo or undefined)). Just getting `undefined` itself isn't possible as far as I know, since `.map(...)` always returns an array

> A removed game should not be in the library on restart of heroic

Yes, that's... what I was saying. It's not in `libraries` (since those are fetched dynamically from the games in the users account(s)), but it *is* going to be in `recentGames` (since that's just a config file we write to when starting a game). And this function goes over each game in `recentGames` and tries to find it in `libraries`, while `libraries` isn't guaranteed to have this game anymore. That's how you might get `undefined`, and that's why you need the `.filter(Boolean)` after

Reply:
I don't see the problem actually. What we do is filtering out recent games which are `undefined`, but we do that before we searching the library. The old approach was incorrect because we called `game.appName` which is not possible on `undefined`. Fitlering out all `undefined` recent games and then find all matching games will return a empty `GameInfo[]` array or a `GameInfo[]` array with the found games.

Reply:
Hm, there's something I don't quite understand there. So accessing `game.appName` won't work when `game` is `undefined`, alright, but how is it then fine to return `undefined` as a `GameInfo` object at the end here?  
Can I then do `gameInfoObj.app_name` for example, even if it's `undefined`? And if I can, why couldn't I do `game.appName` earlier if `game` is `undefined`? Isn't that the same situation there?

Reply:
Ok i finally understand it and found the solution. We should not use map, we should use filter. So it will always be a `GameInfo[]`and not a `(GameInfo[] | undefined)[]`

Reply:
If you'd just use `.filter(...)`, wouldn't you return part of `recentGames`, so a `RecentGame[]`, and not a `GameInfo[]`? I think the `.map(...)` is still necessary, you'd just do it after `.filter()`ing

Reply:
Check back if it makes sense now. The mapping was totally wrong. We need to filter over library and not map over recentGames.

Reply:
Should be faster or same speed. Before we iterate over recent and for each recent game over libraries. Now we just iterate once over libraries.

Reply:
Right, I guess before the speed was dependent on where exactly the recent games were in the library array
Anyways, if it helps with readability (and it certainly does), it's fine by me even if it would be a little slower